kyo-combinators/shared/src/main/scala/kyo/KyoCombinators.scala

```diff
@@ -0,0 +1,245 @@
+    def <*[A1, S1](next: => A1 < S1)(using Frame): A < (S & S1) =
+        effect.map(e => next.andThen(e))
+
+    /** Performs this computation and then the next one, returning both results as a tuple.
+      *
+      * @param next
+      *   The computation to perform after this one
+      * @return
+      *   A computation that produces a tuple of both results
+      */
+    @targetName("zip")
+    def <*>[A1, S1](next: => A1 < S1)(using Frame): (A, A1) < (S & S1) =
+        effect.map(e => next.map(n => (e, n)))
+
+    /** Performs this computation and prints its result to the console.
+      *
+      * @return
+      *   A computation that produces the result of this computation
+      */
+    def debugValue(using Frame): A < S = Debug(effect)
+
+    /** Performs this computation and prints its result to the console with a detailed execution trace.
+      *
+      * @return
+      *   A computation that produces the result of this computation
+      */
+    def debugTrace(using Frame): A < S = Debug.trace(effect)
+
+    /** Performs this computation after a delay.
+      *
+      * @param duration
+      *   The duration to delay for
+      * @return
+      *   A computation that produces the result of this computation with Async effect
+      */
+    def delayed(duration: Duration)(using Frame): A < (S & Async) =
+        Async.delay(duration)(effect)
+
+    /** Performs this computation repeatedly with a backoff policy.
+      *
+      * @param policy
+      *   The backoff policy to use
+      * @return
+      *   A computation that produces the result of the last execution
+      */
+    def repeat(schedule: Schedule)(using Flat[A], Frame): A < (S & Async) =
+        Clock.use { clock =>
+            Loop(schedule) { schedule =>
+                clock.now.map { now =>
+                    schedule.next(now).map { (delay, nextSchedule) =>
+                        effect.delayed(delay).andThen(Loop.continue(nextSchedule))
+                    }.getOrElse {
+                        effect.map(Loop.done)
+                    }
+                }
+            }
+        }
+
+    /** Performs this computation repeatedly with a limit.
+      *
+      * @param limit
+      *   The maximum number of times to repeat the computation
+      * @return
+      *   A computation that produces the result of the last execution
+      */
+    def repeat(limit: Int)(using Flat[A], Frame): A < S =
+        Loop.indexed { i =>
+            if i >= limit then effect.map(Loop.done)
+            else effect.andThen(Loop.continue)
+        }
+
+    /** Performs this computation repeatedly with a backoff policy and a limit.
+      *
+      * @param backoff
+      *   The backoff policy to use
+      * @param limit
+      *   The limit to use
+      * @return
+      *   A computation that produces the result of this computation with Async effect
+      */
+    def repeat(backoff: Int => Duration, limit: Int)(using Flat[A], Frame): A < (S & Async) =
```

Comment:
we have similar methods in `Clock.repeat*` now. It'd be nice to follow the same semantics and naming.

Reply:
I'll take a look